cmake/Protobuf.cmake

```diff
@@ -44,20 +44,25 @@ macro(p4c_obtain_protobuf)
       USES_TERMINAL_DOWNLOAD TRUE
       GIT_PROGRESS TRUE
     )
+    # Derive the target architecture in order to download the right zip.
+    set(protobuf_ARCH "x86_64")
```

Comment:
Is it possible to set the value to just `CMAKE_HOST_SYSTEM_PROCESSOR` and throw an error if we encounter an unsupported arch? We should be able to support `x86_64` and `aarch_64` and `arm64`.

Reply:
Hi @fruffy, the problem is that the protoc binary in [Protobuf releases](https://github.com/protocolbuffers/protobuf/releases/tag/v21.11) is called `aarch_64`, that is why I have to remap the `arm64` value coming from `uname -r` to that one.

However, I agree in adding an error when the architecture is not supported.

Reply:
Hi @fruffy,
I added an additional check on the `arm64` value and also the error handling.

Let me know if you need additional changes. 

Thanks.